docs/manual/codespaces.md

```diff
@@ -0,0 +1,273 @@
+
+### Launching the environment
+
+Once you are satisfied with the environment, you can add a nice button to the README file that will allow users to start the environment with a single click.
+
+/// tab | Button
+<div align=center markdown>
+<a href="https://codespaces.new/srl-labs/srlinux-vlan-handling-lab?quickstart=1">
+<img src="https://gitlab.com/rdodin/pics/-/wikis/uploads/d78a6f9f6869b3ac3c286928dd52fa08/run_in_codespaces-v1.svg?sanitize=true" style="width:50%"/></a>
+
+**[Run](https://codespaces.new/srl-labs/srlinux-vlan-handling-lab?quickstart=1) this lab in GitHub Codespaces for free**.  
+[Learn more](https://containerlab.dev/manual/codespaces) about Containerlab for Codespaces.  
+<small>Machine type: 2 vCPU · 8 GB RAM</small>
+</div>
+///
+/// tab | Button code
+**Do not forget to change the lab repo URL and machine type in the code below!**
+
+```html
+---
+<div align=center markdown>
+<a href="https://codespaces.new/srl-labs/srlinux-vlan-handling-lab?quickstart=1">
+<img src="https://gitlab.com/rdodin/pics/-/wikis/uploads/d78a6f9f6869b3ac3c286928dd52fa08/run_in_codespaces-v1.svg?sanitize=true" style="width:50%"/></a>
+
+**[Run](https://codespaces.new/srl-labs/srlinux-vlan-handling-lab?quickstart=1) this lab in GitHub Codespaces for free**.  
+[Learn more](https://containerlab.dev/manual/codespaces) about Containerlab for Codespaces.  
+<small>Machine type: 2 vCPU · 8 GB RAM</small>
+</div>
+
+---
+```
+
+///
+
+Check out [srl-labs/srl-streaming-telemetry](https://github.com/srl-labs/srl-telemetry-lab?tab=readme-ov-file#nokia-sr-linux-streaming-telemetry-lab) README where this button is used to start the lab in Codespaces.
+
+And of course, you can always launch the Codespace using the GitHub UI by clicking on the "Code" button.
+
+![start](https://gitlab.com/rdodin/pics/-/wikis/uploads/34a9e417d61c02cf9a4283cc1c2cf8cb/image.png)
+
+1. Click on the "Code" button in the GitHub UI.
+2. Start the Codespace on the `main` branch using the devcontainer settings defined in the `.devcontainer/devcontainer.json` file.
+3. Or open up an advanced menu
+4. And configure the Codespace settings manually.
+
+## Dev Container
+
+The key pillar behind Codespaces is the Containerlab' Dev Container image that defines the environment in which the lab will run. The Dev Container image is a Docker image that contains all the necessary tools and dependencies to run Containerlab and other tools that you might need in the lab.
+
+The following two files define Containerlab's Dev Container image:
+
+1. [devcontainer.json](https://github.com/srl-labs/containerlab/blob/main/.devcontainer/devcontainer.json) - the Dev Container configuration file that defines how the environment is built, configured and launched.
+2. [Dockerfile](https://github.com/srl-labs/containerlab/blob/main/.devcontainer/Dockerfile) - the Dockerfile that the Dev Container is built from.
```

Comment:
Dockerfile is not mandatory part of dev container. The core of dev containers are pre-build images, features and dev container spec. May be mention Dockerfile as an option, I think we have to change it later anyway. Refer to dev container spec somewhere: https://containers.dev/implementors/spec/

Reply:
This section is about Containerlab' dev container setup, which uses the Dockerfile

Reply:
yes, thanks, done in [31c91f8](https://github.com/srl-labs/containerlab/pull/2089/commits/31c91f863447354e3c33333d28d4ffce1fab412e)